pkg/kubelet/allocation/allocation_manager.go

```diff
@@ -114,6 +138,94 @@ func (m *manager) UpdatePodFromAllocation(pod *v1.Pod) (*v1.Pod, bool) {
 	return updatePodFromAllocation(pod, allocs)
 }
 
+func GetPodResizeAllocation(pod *v1.Pod) (*PodResourceSummary, bool) {
+	resizeManager.mutex.RLock()
+	defer resizeManager.mutex.RUnlock()
+
+	resizeAllocation, exists := resizeManager.podResizeAllocation[pod.UID]
+	if !exists {
+		return nil, false
+	}
+	return resizeAllocation, true
+}
+
+func ClearPodResizeAllocation(pod *v1.Pod) {
+	resizeManager.mutex.RLock()
+	defer resizeManager.mutex.RUnlock()
+	delete(resizeManager.podResizeAllocation, pod.UID)
+}
+
+func setPodResizeAllocation(pod *v1.Pod, allocationsummary *PodResourceSummary) {
+	resizeManager.mutex.RLock()
+	defer resizeManager.mutex.RUnlock()
+
+	resizeManager.podResizeAllocation[pod.UID] = allocationsummary
+}
+
+// Compare reqOld and reqNew, if different, store reqNew resource to diffReq
+func compareAndSaveDiff(reqOld, reqNew v1.ResourceRequirements) v1.ResourceRequirements {
+	var diffReq v1.ResourceRequirements
+
+	// Compare Requests resource
+	diffReq.Requests = make(v1.ResourceList)
+	for key, reqOldVal := range reqOld.Requests {
+		if reqNewVal, exists := reqNew.Requests[key]; !exists || !reqNewVal.Equal(reqOldVal) {
+			diffReq.Requests[key] = reqOldVal
```

Comment:
This line still stores old value in the diffReq.
The comment on line 165 says: `Compare reqOld and reqNew, if different, store reqNew resource to diffReq`
Should it store reqNew or the comment is incorrect?